### DISCUSSION TITLE
Antennapod auto-download says "host's server sent a broken podcast link" after switching from Podgrab to Podfetch

### DISCUSSION THREAD

**Discussion by Ethanol6** [SamTV12345/PodFetch#455] [Q&A]
"appears to have added the same podcast twice" says the error in Antennapod.

I was using Podgrab as the source for Antennapod but switched over, finally, to Podfetch. I get this error now.

I tried to switch back to the Podgrab rss feed and the error continues. Is anyone else having an issue like this? I can manually download any episode of a podcast but the auto-download fails with the errors listed in the title.

**Comment by SamTV12345** [comment]
Weird. I restructured the api, so it should work now better. Can you try the latest develop image? Hopefully that fixes the issue.

**Reply by Ethanol6** [reply]
I'm unsure what fixed the issue. Some combination of deleting the cache, removing the app and then restarting the phone. Some sort of residual files that weren't cleaned up until I rebooted. It is working now.

**Comment by Ethanol6** [comment]
This is actually still an issue. I've tried Podgrab, PodFetch and even the original URLs for my podcasts and it's the same issue for every podcast. It appears to be not an issue with PodFetch specifically.

**Reply by SamTV12345** [reply]
Hm. If this occurs with any backend this sounds like a bug in AntennaPod. Do you have a reproducible step by step guide? I'd be happy to tinker around with it.

**Reply by Ethanol6** [reply]
This started happening when I remade my server last week. Ubuntu server 22.04. During installation you can choose to install Docker. I didn't realise this was the snap version of Docker until a few days ago when apparmor started causing problems with starting and stopping containers. I removed the snap, installed Docker with apt and remade my containers. 

It's as if there is a duplicate container still going, hosting podcasts at the exact same address, from the previous Docker snap installation. `docker ps` doesn't reveal anything strange.

**Reply by SamTV12345** [reply]
Normally this shouldn't be possible as there can only be 1 application running on a port. I have no idea how this happened. Can you try to reproduce that with a podcast?

**Comment by Ethanol6** [comment]
Sorry for the delay. This appears to be a misunderstanding on my part. Some of my podcasts had this error on some episodes which appear to be a combination of an issue with the podcast itself and Antennapod's duplicate episode filtering which is convinced that some episodes which only differ by a single episode number (podcast-name-1234 and podcast-name-1235 for example) are the same episode. 

The other part about the auto-download not working is a separate PodFetch issue. Since podcasts are updated automatically when you visit the podcast page, applications like Antennapod will never receive an updated episode list until you go to the page for each podcast manually. In my case, Antennapod was never going to auto-download new episodes because it was never receiving new episodes.

Thanks for your assistance.
